src/Components/OATHeader/OATHeader.tsx

```diff
@@ -316,34 +313,55 @@ const OATHeader = ({ dispatch, state }: OATHeaderProps) => {
     };
 
     const handleFileListChanged = async (files: Array<File>) => {
-        const items = [];
+        const newModels = [];
         if (files.length > 0) {
             const filesErrors = [];
             let modelsMetadataReference = null;
             for (const current of files) {
                 const content = await current.text();
-                const error = await parseModel(content);
-                if (!error) {
+                const JSONContent = JSON.parse(content);
+                newModels.push(JSONContent);
+            }
+
+            const combinedModels = [...models, ...newModels];
+            const error = await parseModels(combinedModels);
+
+            if (!error) {
+                console.log('No error');
+                let loopIndex = 0;
+                for (const current of files) {
                     modelsMetadataReference = populateMetadata(
                         current,
-                        content,
+                        JSON.stringify(newModels[loopIndex]),
                         modelsMetadataReference
                     );
-
-                    items.push(JSON.parse(content));
-                } else {
-                    filesErrors.push(
-                        t('OATHeader.errorIssueWithFile', {
-                            fileName: current.name,
-                            error
-                        })
-                    );
+                    loopIndex++;
                 }
+            } else {
+                filesErrors.push(
+                    t('OATHeader.errorIssueWithFile', {
+                        fileName: t('OATHeader.file'),
+                        error
+                    })
+                );
             }
+
             if (filesErrors.length === 0) {
+                const modelsCopy = deepCopy(models);
+                // Add new models to models copy
+                for (const model of newModels) {
+                    // Check if model already exists
+                    const modelExists = modelsCopy.find(
+                        (m) => m['@id'] === model['@id']
+                    );
+                    if (!modelExists) {
```

Comment:
I think we should do this check after parsing. I.e. this is actually an error and we should add a message to the errors collection and stop.